fastpickle.py

```diff
@@ -488,6 +492,193 @@ def dump(self, obj):
         self.write(STOP)
         self.framer.end_framing()
 
+    def pardumps(self, obj, pickling_order):
+        """ Customized dumps() for parallelization. 
+            obj : list of [list | set | dict]
+            pickling_order : list containing the order in which the children of obj will be pickled
+            
+            Returns :
+            a bytecode such that pickle.dumps(obj) == pickle.pardumps(obj)
+        """
+        
+        if not hasattr(self, "_file_write"):
+            raise PicklingError("Pickler.__init__() was not called by "
+                                "%s.__init__()" % (self.__class__.__name__,))
+        if self.proto >= 2:
+            self.write(PROTO + pack("<B", self.proto))
+        if self.proto >= 4:
+            self.framer.start_framing()
+
+        assert len(obj) == len(pickling_order)
+
+        # dummy element, -1 can never be the id of an object
+        self.memo[-1] = None
+
+        self.pickling_order = pickling_order
+        self.child_bytes = [0] * len(obj)
+        for i, v in enumerate(pickling_order):
+            self.thread_num = v
+            sp = self.framer.current_frame.tell()
+            self.save(obj[v])
+            self.child_bytes[v] = self.framer.current_frame.getvalue()[sp:]
+        
+        print("self.child_bytes",self.child_bytes)
+        print("self.threads_access", self.threads_access)
+        print("self.memo",self.memo)
+        print("self.local_memo", self.local_memo)
```

Comment:
Let's get in the habit of using `logging.debug` instead of `print`, if you want to keep these.

Debugging `print` is fine for local dev but not in commit. You can comment them too